Use view binding in stats

planarvoid opened the pull request:
This PR introduces ViewBindings in the stats, stats detail and stats view all screens. There were several tricky things to figure out. 
- When using the FragmentContainerView we now can't initialize the activity action bar in activity `onCreate` so I've moved it to fragment `onViewCreated`. 
- The `@SuppressLint("ClickableViewAccessibility")` is there because the way we catch the touches on the view pager makes the AS think that it affects accessibility. As far as I could see this is not the case so I've added the suppression. Let me know if you think otherwise. 
- I'm not sure my approach to `binding` is the best. I'm trying to pass it around as a parameter as much as possible and only use the field when the binding needs to be accessible from other methods than `onViewCreated`. When using `views` this is the safest approach because you can be sure the view is there. Here I'm not 100% sure.

To test:
- Go to Stats and notice it's loaded correctly
- Notice the action bar has the correct title and the back button is active
- Click on `View more` on a block on a site with a lot of data (you might need to go to the `Year` view)
- Notice the "View all" screen is loaded correctly
- Notice the action bar has the correct title and the back button is active
- Go to My Site/Blog posts
- Click on the overflow menu on one of the older blog posts (not an X-post) and go to "Post Stats"
- Notice the screen is loaded correctly
- Notice the action bar has the correct title and the back button is active

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

planarvoid committed:
View binding in Stats activity

planarvoid committed:
Use view binding in StatsFragment.kt

planarvoid committed:
Use view binding in StatsViewAllActivity.kt

planarvoid committed:
Move actionbar setup to fragments

planarvoid committed:
Use `binding` field instead of binding multiple times

planarvoid committed:
Address detekt issues

planarvoid committed:
Merge branch 'feature/add-observe-event-method' into feature/use-view-biding-in-stats

renanferrari commented on WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsFragment.kt:
```diff
@@ -38,28 +39,37 @@ import javax.inject.Inject
         super.onViewCreated(view, savedInstanceState)
 
         val nonNullActivity = requireActivity()
+        binding = StatsFragmentBinding.bind(requireView())
```
I believe we can use the parameter `view` here.
 
```suggestion
        binding = StatsFragmentBinding.bind(view)
```

renanferrari commented on WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsFragment.kt:
```diff
@@ -38,28 +39,37 @@ import javax.inject.Inject
     @Inject lateinit var uiHelpers: UiHelpers
     private lateinit var viewModel: StatsViewModel
     private lateinit var swipeToRefreshHelper: SwipeToRefreshHelper
+    private lateinit var binding: StatsFragmentBinding
```
I'm not sure if we should make the `binding` a `lateinit var` on fragments. The best practice seems to be to initialize [like this](https://developer.android.com/topic/libraries/view-binding#fragments) and then clear it on `onDestroyView()` to avoid memory leaks. That said, I've used this approach on #14224 and I'm not so sure I like it. I'm 100% open to other suggestions.

planarvoid replied:
you're right, I've missed this part 👍 . I'll change the variables

renanferrari reviewed (commented):
Thanks for the changes! I've tested the app and everything seems to be working as expected. Code is looking good overall but I'm also not sure about the approach to `binding`. I agree that passing it around as a parameter is the best option whenever possible, but as for when it needs to be accessed from other methods I'm still not sure what's the best approach as well.

planarvoid committed:
Update WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsFragment.kt

Co-authored-by: Renan Ferrari <renan.fb@gmail.com>

planarvoid committed:
Properly clear view binding field

planarvoid commented:
good catch @renanferrari . I've changed the logic around binding a bit. What do you think about my changes? Instead of passing the binding as a parameter the methods are now in binding scope. I've also added clearing for the binding field.

planarvoid committed:
Fix detekt issue

renanferrari commented on WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsViewAllFragment.kt:
```diff
@@ -89,35 +84,52 @@ class StatsViewAllFragment : DaggerFragment() {
         super.onViewCreated(view, savedInstanceState)
 
         val nonNullActivity = requireActivity()
+        with(StatsViewAllFragmentBinding.bind(requireView())) {
```
Sorry, I missed this one the first time!

```suggestion
        with(StatsViewAllFragmentBinding.bind(view)) {
```

renanferrari reviewed (approved):
>Instead of passing the binding as a parameter the methods are now in binding scope.

I really liked that! Looks really clean 👌

Feel free to merge this once you addressed the suggestion below.

LGTM 👍

planarvoid committed:
Update WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsViewAllFragment.kt

Co-authored-by: Renan Ferrari <renan.fb@gmail.com>
